src/host/VtIo.cpp

```diff
@@ -567,17 +567,26 @@ void VtIo::Writer::WriteUTF16(std::wstring_view str) const
 
     // C++23's resize_and_overwrite is too valuable to not use.
     // It reduce the CPU overhead by roughly half.
-#if !defined(_HAS_CXX23) || !_HAS_CXX23
-#define resize_and_overwrite _Resize_and_overwrite
+#if !defined(__cpp_lib_string_resize_and_overwrite)

```

Comment:
Using feature test macros instead of MSVC private macros; this will no longer be necessary after upgrading to C++23.

Reply:
I think the change towards `__cpp_lib_string_resize_and_overwrite`, but I don't believe the change away from `#define resize_and_overwrite _Resize_and_overwrite` is useful. It's just more code that needs to be read, understood, and eventuelly removed once we use C++23.